grpc/v1/webserver.pb.go

```diff
@@ -0,0 +1,4728 @@
+	Priority float64 `protobuf:"fixed64,1,opt,name=priority,proto3" json:"priority,omitempty"`
+	// max_pending_priority is the maximum priority of pending result that is being searched in parallel.
+	// This is used to reorder results when the result set is known to be stable-- that is, when a result's
+	// Priority is greater than the max(MaxPendingPriority) from the latest results of each backend, it can be returned to the user.
+	//
+	// max_pending_priority decreases monotonically in each SearchResult.
+	MaxPendingPriority float64 `protobuf:"fixed64,2,opt,name=max_pending_priority,json=maxPendingPriority,proto3" json:"max_pending_priority,omitempty"`
+}
+
+func (x *Progress) Reset() {
+	*x = Progress{}
+	if protoimpl.UnsafeEnabled {
+		mi := &file_webserver_proto_msgTypes[32]
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		ms.StoreMessageInfo(mi)
+	}
+}
+
+func (x *Progress) String() string {
+	return protoimpl.X.MessageStringOf(x)
+}
+
+func (*Progress) ProtoMessage() {}
+
+func (x *Progress) ProtoReflect() protoreflect.Message {
+	mi := &file_webserver_proto_msgTypes[32]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
+	}
+	return mi.MessageOf(x)
+}
+
+// Deprecated: Use Progress.ProtoReflect.Descriptor instead.
+func (*Progress) Descriptor() ([]byte, []int) {
+	return file_webserver_proto_rawDescGZIP(), []int{32}
+}
+
+func (x *Progress) GetPriority() float64 {
+	if x != nil {
+		return x.Priority
+	}
+	return 0
+}
+
+func (x *Progress) GetMaxPendingPriority() float64 {
+	if x != nil {
+		return x.MaxPendingPriority
+	}
+	return 0
+}
+
+// FileMatch contains all the matches within a file.
+type FileMatch struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	// Ranking; the higher, the better.
+	Score float64 `protobuf:"fixed64,1,opt,name=score,proto3" json:"score,omitempty"`
+	// For debugging. Needs DebugScore set, but public so tests in
+	// other packages can print some diagnostics.
+	Debug    string `protobuf:"bytes,2,opt,name=debug,proto3" json:"debug,omitempty"`
+	FileName string `protobuf:"bytes,3,opt,name=file_name,json=fileName,proto3" json:"file_name,omitempty"`
+	// Repository is the globally unique name of the repo of the
+	// match
+	Repository string   `protobuf:"bytes,4,opt,name=repository,proto3" json:"repository,omitempty"`
+	Branches   []string `protobuf:"bytes,5,rep,name=branches,proto3" json:"branches,omitempty"`
+	// One of line_matches or chunk_matches will be returned depending on whether
+	// the SearchOptions.ChunkMatches is set.
+	LineMatches  []*LineMatch  `protobuf:"bytes,6,rep,name=line_matches,json=lineMatches,proto3" json:"line_matches,omitempty"`
+	ChunkMatches []*ChunkMatch `protobuf:"bytes,7,rep,name=chunk_matches,json=chunkMatches,proto3" json:"chunk_matches,omitempty"`
```

Comment:
If I remember when you added these fields we talked about the impact on the garbage collector allocating all these pointers would have. Is this not a concern now?

Reply:
It definitely is still a concern, but is also a pretty fundamental limitation of Go's implementation of protobuf. Message types must be pointers 🙃 

If this shows up as significant in profiles, there are a few options. Probably first would be to try flattening the message type. We could also try to do some object pooling. I've seen some interesting (and scary) posts about using `runtime.SetFinalizer()` to reuse message allocations. 

Worst case scenario, we can always drop proto and use a different encoding.

Since it's not trivial to make better, my take is "wait for the profiles to tell us to fix it."

Reply:
ok cool, lets just be proactive about it then and make sure we check on dot-com before and after. SGTM